### PR TITLE
Ui update search term

### PR DESCRIPTION
**Pull request by Matt-Warnock**

### CODE REVIEW

File: spec/user_interface_spec.rb [hunk 1]
```diff
@@ -221,6 +220,67 @@
...
+
+    it 'returns the search term typed by user' do
+      input = StringIO.new("john\n")
+      ui = described_class.new(input, output, validator)
+
+      expect(ui.search_term).to eq('john')
+    end
+
+    it 'prints error message if no input is given' do
+      input = StringIO.new("\njohn\n")
```
Comment (octopusinvitro) [inline spec/user_interface_spec.rb:241]: according to this test description ("...if no input is given" ) you just need:

```ruby
input = StringIO.new("\n")
```
nope?

File: spec/user_interface_spec.rb [hunk 2]
```diff
@@ -221,6 +220,67 @@
...
+
+    it 'returns true if user wants to search another contact' do
+      input = StringIO.new(yes_reply)
+      ui = described_class.new(input, output, validator)
+
+      expect(ui.search_again?).to eq(true)
+    end
+
+    it 'returns false if user doesnt want to search another contact' do
+      input = StringIO.new('n')
```
Comment (octopusinvitro) [inline spec/user_interface_spec.rb:268]: doesn't this need a `\n` at the end to represent reality? 

until the user presses enter, the method `gets` can not read the value, so in this test `search_again?` is false just by accident.

**Review by octopusinvitro — APPROVED** [review]
Good work, just refactoring comments
 
(I approved but would like you to address the feedback)

File: lib/user_interface.rb [hunk 3]
```diff
@@ -76,7 +78,7 @@ def display_no_contacts_message
 
   def add_another_contact?
     output.print ANOTHER_CONTACT_PROMPT
-    collect_vaild_input { |user_input| validator.valid_yes_no_answer?(user_input) }.downcase == YES_REPLY
+    collect_vaild_input { |user_input| validator.valid_yes_no_answer?(user_input) } == YES_REPLY
```
[the remark this answers is not in the document]
Comment (Matt-Warnock) [reply lib/user_interface.rb:81]: Because I've got completely confused combined with poor coverage in my test

**Pull request merged**